### PR TITLE
Fix broken links, grammar, indentation in README.md

### PR DESCRIPTION
**Pull request by renormalize**
**What this PR does / why we need it**:

* Fix the broken link to install Kustomize.

* Fix a broken link to clusterpodschedulingpolicy in "support for top-down pod scheduling criteria".

* Fix grammatical errors, typos.

* Adhere to better markdown practices overall.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,24 +62,29 @@ This is especially relevant for supporting dedicated worker pools for shoot `etc
 
 The steps for installing kupid on a Kubernetes cluster for development and/or trial are given below.
 These are only development installation steps and not intended for any kind of production scenarios.
-For anything other than development or trial purposes, please use your favourite CI/CD toolkit.
+For anything other than development or trial purposes, please use your favorite CI/CD toolkit.
```
Comment (shreyas-s-rao) [inline README.md:65]: Not really necessary to change this, since it's just a difference between American and British spelling.

**Review by shreyas-s-rao — APPROVED** [review]
@renormalize thanks for the PR. If you read the [Kyverno Evaluation document](https://github.com/gardener/kupid/blob/master/docs/Kyverno%20Evaluation.md), you would know that the plan is to replace Kupid with Kyverno soon, so the time you put in for this may not have been necessary 😅 .

Anyway, thanks for making the changes and updating this document.

/lgtm

File: README.md (hunk 1 above)
Comment (renormalize) [reply README.md:65]: Yes that's true - the quick fix tool that was used to fix grammar autocorrected it and I didn't notice this particular change. Sorry about that.

**Comment by renormalize** [comment]
Thanks @shreyas-s-rao for the review! I did take a look at the Kyverno Evaluation doc - but it wasn't really any effort to fix this document. It's always nicer to have better docs, even for a project that is planned to be sunset in the near future.

**Pull request merged**